parts/k8s/kubernetescustomscript.sh

```diff
@@ -62,7 +62,7 @@ else
 fi
 
 function testOutboundConnection() {
-    retrycmd_if_failure 20 1 3 nc -v 8.8.8.8 53 || retrycmd_if_failure 20 1 3 nc -v 8.8.4.4 53 || exit $ERR_OUTBOUND_CONN_FAIL
+    retrycmd_if_failure 20 1 3 curl -I https://www.google.com || retrycmd_if_failure 20 1 3 curl -I https://www.1688.com || exit $ERR_OUTBOUND_CONN_FAIL
```

Comment:
A cleaner implementation of the intent would be `nc -v google.com 443` (same for the alibaba test)

Reply:
@jackfrancis perhaps, however, the "curl https://" prefix also tests DNS name resolution, which has been an issue I have had to resolve for corporate clients before they could pull containers from remote repositories.

Reply:
`nc -v google.com` will test name resolution

Reply:
@jackfrancis you are, of course, correct.